src/overlays/actors/ovl_En_Fish2/z_en_fish2.c

```diff
@@ -22,95 +65,1112 @@ const ActorInit En_Fish2_InitVars = {
+        if (this->actor.speedXZ > 3.0f) {
+            this->actor.speedXZ = 3.0f;
+        } else if (this->actor.speedXZ < 1.5f) {
+            this->actor.speedXZ = 1.5f;
+        }
+    } else {
+        Math_ApproachZeroF(&this->actor.speedXZ, 0.3f, 0.3f);
+    }
+
+    if ((D_80B2B2E8 == 0) && (D_80B2B2E0 != 2)) {
+        while (itemAction != NULL) {
+            if ((itemAction->id != ACTOR_EN_FISH) && (itemAction->id != ACTOR_EN_MUSHI2)) {
+                itemAction = itemAction->next;
+                continue;
+            }
+
+            if (itemAction->id == ACTOR_EN_MUSHI2) {
+                if ((itemAction->update != NULL) && (((EnMushi2*)itemAction)->unk_30C & 0x200)) {
+                    this->unk_350 = itemAction;
+                    this->unk_2C8 = 1;
+                    func_80B29194(this);
+                    break;
+                }
+            } else if ((itemAction->update != NULL) && (itemAction->params == 0) &&
+                       (fabsf(itemAction->world.pos.x - this->actor.world.pos.x) < 100.0f) &&
+                       (fabsf(itemAction->world.pos.z - this->actor.world.pos.z) < 100.0f) &&
+                       (itemAction->bgCheckFlags & 0x20)) {
+                this->unk_350 = itemAction;
+                if (D_80B2B2E0 == 0) {
+                    EnFish2* fish;
+
+                    this->unk_354 = NULL;
+                    fish = (EnFish2*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FISH2,
+                                                        this->unk_324.x, this->unk_324.y, this->unk_324.z, 0, 0, 0, 1);
+                    this->unk_354 = fish;
+                    if (this->unk_354 != NULL) {
+                        D_80B2B2E4 = 0;
+                        D_80B2B2E0 = 1;
+                        fish->unk_2BA[0] = this->unk_2BA[0];
+                        fish->unk_350 = this->unk_350;
+                    }
+                }
+                func_80B29194(this);
+                break;
+            }
+            itemAction = itemAction->next;
+        }
+
+        func_800B8614(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void func_80B29128(EnFish2* this) {
+    this->actionFunc = func_80B2913C;
+}
+
+void func_80B2913C(EnFish2* this, GlobalContext* globalCtx) {
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        func_801477B4(globalCtx);
+        func_80B28B5C(this);
+    }
+}
+
+void func_80B29194(EnFish2* this) {
+    if ((this->unk_354 != NULL) && (this->unk_2C8 == 0)) {
+        EnMushi2* mushi = (EnMushi2*)this->unk_350;
+        mushi->unk_277 = 1;
+    }
+
+    this->unk_2C4 = 0;
+    this->actor.speedXZ = 0.0f;
+    if (this->unk_2C8 == 0) {
+        this->unk_34C = 400;
+        Math_Vec3f_Copy(&this->actor.world.pos, &this->unk_324);
+        this->actor.world.pos.y = this->actor.floorHeight + (this->unk_330 * 1200.0f);
+    }
+
+    this->unk_34A = 0;
+    this->unk_340 = (s32)Rand_ZeroOne() & 1;
```

Comment:
I think this would make a bit more of sense with a modulo. 
```suggestion
    this->unk_340 = (s32)Rand_ZeroOne() % 2;
```
Only if it matches tho

Reply:
Doesn't match unfortunately.